File: backend/src/main/java/com/kaamchahiye/app/service/AttendanceService.java
package com.kaamchahiye.app.service;

import com.kaamchahiye.app.dto.request.AttendanceRequest;
import com.kaamchahiye.app.dto.response.AttendanceDto;
import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {
    AttendanceDto recordAttendance(AttendanceRequest request);
    List<AttendanceDto> getAttendanceByWorker(Long workerId);
    List<AttendanceDto> getAttendanceByDate(LocalDate date);
}

File: frontend/src/services/attendanceService.js
import apiClient from './apiClient';

export const attendanceService = {
  recordAttendance: async (attendanceData) => {
    const response = await apiClient.post('/attendance', attendanceData);
    return response.data;
  },

  getAttendanceByWorker: async (workerId) => {
    const response = await apiClient.get(`/attendance/worker/${workerId}`);
    return response.data;
  },

  getAttendanceByDate: async (date) => {
    const response = await apiClient.get('/attendance/date', { params: { date } });
    return response.data;
  }
};

export default attendanceService;
